servers/quarkus-server/src/main/java/org/projectnessie/server/catalog/CatalogProducers.java

```diff
@@ -284,9 +286,13 @@ public Executor importJobExecutor(ThreadContext threadContext, CatalogServiceCon
   }
 
   @Produces
-  @Singleton
-  public NessieApiV2 nessieApiV2(Persist persist) {
-    return new CombinedClientBuilder().withPersist(persist).build(NessieApiV2.class);
+  @RequestScoped
+  public NessieApiV2 nessieApiV2(
+      RestV2ConfigResource configResource, RestV2TreeResource treeResource) {
+    return new CombinedClientBuilder()
+        .withConfigResource(configResource)
+        .withTreeResource(treeResource)
+        .buildForProduction(NessieApiV2.class);
```

Comment:
Do we really need the new method? Calling the old `build` method would yield the same result.

Reply:
Wanna give that another look separately. Potentially removing the code path in `build()` if it's not used anywhere.